### ISSUE TITLE
ActionBar: Esc key to close ActionGroupMenu within ActionBar clears selection and closes ActionBar

### ISSUE THREAD

**Issue by majornista** [adobe/react-spectrum#2449]
# 🐛 Bug Report 

At https://github.com/adobe/react-spectrum/blob/2731ebc7d1992b167e3bbab6171e28f2b0adb7fa/packages/%40react-spectrum/actionbar/src/ActionBar.tsx#L71-L78,

The Esc key to close the ActionBar always clears selection. So if the Esc key is triggered from within the ActionGroupMenu to close the Menu of action items that don't fix within the ActionBar, the current selection will be lost, rather than maintaining selection and restoring focus the the ActionGroup menu trigger.

Also, unlike closing the ActionBar by clicking the Dismiss button, pressing the Esc key does not restore focus to the TableView, but instead focus is lost to the `document.body`.

## 🤔 Expected Behavior

Esc key handler should first check that the current focus is actually within the ActionBar, and not within the Portal containing the ActionGroupMenu, before clearing the selection.

When selection is cleared using the Esc key, focus should be restored to the TableView, this may relate to #2444.

## 😯 Current Behavior

The Esc key to close the ActionBar always clears selection. So if the Esc key is triggered from within the ActionGroupMenu to close the Menu of action items that don't fix within the ActionBar, the current selection will be lost, rather than maintaining selection and restoring focus the the ActionGroup menu trigger.

## 💁 Possible Solution

Add a test to make sure the target for the Esc keyboard event is a descendant of the ActionBar before clearing selection.

## 🌍 Your Environment

| Software         | Version(s) |
| ---------------- | ---------- |
| react-spectrum   |  @react-spectrum/actionbar v3.0.0-alpha.3
| Browser          | All
| Operating System | All

## 🧢 Your Company/Team

Adobe/Accessibility
